tools, README: add DM config file docs (#779)
pingcap/docs: tools/dm-task-configuration-file-intro.md

            host: "192.168.199.118"
            port: 4306
            user: "root"
            password: "1234"                       # Requires the password encrypted by dmctl
        instance-id: "instance118-4306"            # The MySQL instance ID. It corresponds to the upstream MySQL instance. It is 
                                                   # not allowed to set it to an ID of a MySQL instance that is not within the 
                                                   # DM-master cluster topology.

        meta:                                      # The position where the binlog synchronization starts when the checkpoint of 
                                                   # the downstream database does not exist. If the checkpoint exits, this 
                                                   # configuration does not work. 
            binlog-name: binlog-00001
            binlog-pos: 4

        route-rules: ["user-route-rules-schema", "user-route-rules"]       # Routing rules selected from `routes` above
        filter-rules: ["user-filter-1", "user-filter-2"]                   # Filter rules selected from `filters` above

>         column-mapping-rules: ["instance-1"]                               # The column mapping rule name of the matched table of 

Reviewer: this comment can also be:
Column mapping rules selected from `column-mappings` above

    
        loader:                                                            # The Loader configuration. You cannot set it and 
                                                                           # `loader-config-name` at the same time.
            pool-size: 32                                                  # The number of threads that execute mydumper SQL 
                                                                           # files concurrently in Loader
            dir: "./dumped_data"
    
        syncer:                                                            # The Syncer configuration. You cannot set it and 
                                                                           # `syncer-config-name` at the same time.
            worker-count: 32                                               # The number of threads that synchronize binlog events 
                                                                           # concurrently in Syncer
            batch: 2000
            max-retry: 200
```

Refer to [Feature configuration set](#feature-configuration-set) to get the configuration details of the following options:

> - `route-rules`

Reviewer 2: To make it clearer, (for these options in the list) please specify which option the users should refer to in "Feature configuration set".  Ref: https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/configuration/configuration.md

---

> title: Task Configuration File of Data Migration

Reviewer 2: ```suggestion
title: Data Migration Task Configuration File
```

summary: This document introduces the task configuration file of Data Migration. 
category: tools
---

> # Task Configuration File of Data Migration

Reviewer 2: ```suggestion
# Data Migration Task Configuration File
```

Reviewer 2: To keep the title consistent with that of another config file.

This document introduces the task configuration file of Data Migration --
[`task.yaml`](https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/configuration/task.yaml), including [Global configuration](#global-configuration) and [Instance configuration](#instance-configuration).

> For description of configuration items, see [Task Configuration Argument Description](../tools/dm-task-config-argument-description.md).

Reviewer 2: ```suggestion
For description of configuration items, see [Data Migration Task Configuration Options](../tools/dm-task-config-argument-description.md).
```

## Important concepts

For description of important concepts including `instance-id` and the DM-worker ID, see [Description of important concepts](../tools/dm-configuration-file-overview.md#description-of-important-concepts) 

## Global configuration

### Basic information configuration

```
name: test                      # The name of the task. Should be globally unique.
task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
is-sharding: true               # Whether it is a sharding task
meta-schema: "dm_meta"          # The downstream database that stores the `meta` information

> remove-meta: false              # Whether to remove the `meta` information (`checkpoint` and `onlineddl`) before the 

Reviewer: Whether to remove the `meta` information (`checkpoint` and `onlineddl`) before starting the  synchronization task?

    ​    table-pattern: "t_*"
    ​    expression: "partition id"
    ​    source-column: "id"
    ​    target-column: "id"
    ​    arguments: ["1", "test_", "t_"]

mydumpers:                                          # Configuration arguments of running mydumper
    global:
    ​    mydumper-path: "./mydumper"                 # The mydumper binary file path. It is generated by the Ansible deployment                                                    # application automatically and needs no configuration.
    ​    threads: 16                                 # The number of the threads mydumper dumps from the upstream database instance
    ​    chunk-filesize: 64                          # The size of the file mydumper generates
    ​    skip-tz-utc: true						
    ​    extra-args: "-B test -T t1,t2 --no-locks"

loaders:                                            # Configuration arguments of running Loader
    global:
    ​    pool-size: 16                               # The number of threads that execute mydumper SQL files concurrently in Loader
    ​    dir: "./dumped_data"                        # The directory output by mydumper that Loader reads. Directories for

>                                                     # different tasks of the same instance are different. (mydumper outputs the 

Reviewer: ```suggestion
                                                    # different tasks of the same instance must be different. (mydumper outputs the 
```

```
mysql-instances:
    -
    ​    config:                                    # The upstream database configuration corresponding to `instance-id`
    ​        host: "192.168.199.118"
    ​        port: 4306
    ​        user: "root"
    ​        password: "1234"                       # Requires the password encrypted by dmctl
    ​    instance-id: "instance118-4306"            # The MySQL instance ID. It corresponds to the upstream MySQL instance. It is 
                                                   # not allowed to set it to an ID of a MySQL instance that is not within the 
                                                   # DM-master cluster topology.

    ​    meta:                                      # The position where the binlog synchronization starts when the checkpoint of 
                                                   # the downstream database does not exist. If the checkpoint exits, this 
                                                   # configuration does not work. 
    ​        binlog-name: binlog-00001
    ​        binlog-pos: 4

>     ​    route-rules: ["user-route-rules-schema", "user-route-rules"]       # The name of the mapping rule between the matched 

Reviewer 3: I think we can make the comment simpler, like `that route rules selected from routes above `

                                                                           # table of the upstream database instance and the 
                                                                           # downstream database
    ​    filter-rules: ["user-filter-1", "user-filter-2"]                   # The filter rule name of binlog events of the matched 
                                                                           # table for the upstream database instance
    ​    column-mapping-rules: ["instance-1"]                               # The column mapping rule name of the matched table of 
                                                                           # the upstream database instance 

>     ​    black-white-list:  "instance"                                      # The filter rule of the black white list of the 

Reviewer 3: ditto
